src/pages/index.js

```diff
@@ -0,0 +1,299 @@
+  document.removeEventListener("keyup", handleEscape);
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const newCardData = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .postNewCard(newCardData)
+    .then((card) => {
+      const cardElement = getCardElement(card);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton(cardSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      userAvatar.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+      submitBtn.textContent = "Save";
+    });
+}
+
+function getCardElement(data) {
+  console.log("Data in getCardElement:", data);
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_liked");
+
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then(() => {
+        cardLikeBtn.classList.toggle("card__like-btn_liked");
+        // Optional: Update any other UI elements, like the like counter
+      })
+      .catch(console.error);
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    console.log("Card data when delete clicked:", data);
+    console.log("Card ID being selected:", data._id);
+    selectedCardId = data._id;
+    selectedCard = cardElement;
+    openModal(deleteModal);
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  return cardElement;
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
```

Comment:
`  submitBtn.textContent = "Saving...";` is not needed because you do it with `setButtonText` in line 212